Doc: Update rewrite data files spark procedure (#11396)
apache/iceberg: docs/docs/spark-procedures.md

| `max-concurrent-file-group-rewrites` | 5 | Maximum number of file groups to be simultaneously rewritten |
| `partial-progress.enabled` | false | Enable committing groups of files prior to the entire rewrite completing |
| `partial-progress.max-commits` | 10 | Maximum amount of commits that this rewrite is allowed to produce if partial progress is enabled |

> | `partial-progress.max-failed-commits` | value of `partital-progress.max-commits` | Maximum amount of failed commits is allowed before job failure, if partial progress is enabled |

Reviewer: sorry maybe my comment mistake.  let's remove 'is', i think it came from javadoc but doesnt make sense here.

| `rewrite-all` | false | Force rewriting of all provided files overriding other options |
| `max-file-group-size-bytes` | 107374182400 (100GB) | Largest amount of data that should be rewritten in a single file group. The entire rewrite operation is broken down into pieces based on partitioning and within partitions based on size into file-groups.  This helps with breaking down the rewriting of very large partitions which may not be rewritable otherwise due to the resource constraints of the cluster. |
| `delete-file-threshold` | 2147483647 | Minimum number of deletes that needs to be associated with a data file for it to be considered for rewriting |

> | `output-spec-id` | current partition spec id | Desired partition spec ID to be used for rewriting data files. This allows data files to be rewritten with any existing partition spec. |

Reviewer: I thought the original javadoc is well rewritten here as well, how about taking from that:
```
Identifier of the output partition spec.  Data will be reorganized during the rewrite to align with the output partitioning.
```

> | `remove-dangling-deletes` | false | Remove dangling position and equality deletes after rewriting. A delete file is considered dangling if it does not apply to any live data files. Enabling this will generate an additional snapshot. |

Reviewer: This is just my thought, but last sentence could be clarified a little:

'This will generate an additional commit for the removal'

> !!! info

Reviewer: Nit:  should we also add about position deletes?

, nor to position delete files containing position deletes no longer matching any live data files.

| `rewrite-all` | false | Force rewriting of all provided files overriding other options |
| `max-file-group-size-bytes` | 107374182400 (100GB) | Largest amount of data that should be rewritten in a single file group. The entire rewrite operation is broken down into pieces based on partitioning and within partitions based on size into file-groups.  This helps with breaking down the rewriting of very large partitions which may not be rewritable otherwise due to the resource constraints of the cluster. |
| `delete-file-threshold` | 2147483647 | Minimum number of deletes that needs to be associated with a data file for it to be considered for rewriting |

> | `output-spec-id` | current partition spec id | Identifier of the output partition spec.  Data will be reorganized during the rewrite to align with the output partitioning |

Reviewer: Nit: this doc looks inconsistent, but I guess we can do one space to keep with the majority?

> | `remove-dangling-deletes` | false | Remove dangling position and equality deletes after rewriting. A delete file is considered dangling if it does not apply to any live data files. Enabling this will generate an additional commit for the removal |

Reviewer: Nit: period at the end

| `rewrite-all` | false | Force rewriting of all provided files overriding other options |
| `max-file-group-size-bytes` | 107374182400 (100GB) | Largest amount of data that should be rewritten in a single file group. The entire rewrite operation is broken down into pieces based on partitioning and within partitions based on size into file-groups.  This helps with breaking down the rewriting of very large partitions which may not be rewritable otherwise due to the resource constraints of the cluster. |
| `delete-file-threshold` | 2147483647 | Minimum number of deletes that needs to be associated with a data file for it to be considered for rewriting |

> | `output-spec-id` | current partition spec id | Desired partition spec ID to be used for rewriting data files. This allows data files to be rewritten with a custom partition spec. |

Reviewer 2: output-spec-id - may be mention that `it needs to be one of the existing ones ( previous partitions specs)`

Author: How about 
> This allows data files to be rewritten with one of existing partition specs.

> | `output-spec-id` | current partition spec id | Desired partition spec ID to be used for rewriting data files. This allows data files to be rewritten with one of existing partition specs. |

Reviewer 3: ```suggestion
| `output-spec-id` | current partition spec id | Desired partition spec ID to be used for rewriting data files. This allows data files to be rewritten with any existing partition spec. |
```

Author: thank you @RussellSpitzer , updated as you suggested

> | `remove-dangling-deletes` | false | Remove dangling position and equality deletes after rewriting. A delete file is considered dangling if it does not apply to any live data files. Enabling this will generate an additional snapshot of the delete type. |

Reviewer 4: > Enabling this will generate an additional snapshot of the delete type.

1/  if this commit fails does the whole job fails ? 
2/ can you please elaborate on what does `snapshot of delete type` means from what i am understand the operation of this update should be replace as we use `table.newRewrite()` API

Reviewer 3: Agreed, we don't have a "delete" snapshot type.

Author: yeah I forgot that I switch to newRewrite instead and I think i can reword it better without mention snapshot type.

With respect to remove dangling delete commit, if there's any runtime failure it can fail the whole job but it will not undo the previous rewrite/compaction. User can either investigate to understand the failure or turn off this feature flag to bypass. Dangling delete removal can also be executed as a standalone SparkAction if needed

> | `remove-dangling-deletes` | false | Remove dangling position and equality deletes after rewriting. A delete file is considered dangling if it does not apply to any live data files. Enabling this will generate an additional snapshot of the delete type. |

Reviewer 3: I think this is already pretty long for the description. We could write this up in another section and link to it.

> | `remove-dangling-deletes` | false | Remove dangling position and equality deletes after rewriting. A delete file is considered dangling if it does not apply to any live data files. Enabling this will generate an additional snapshot of the delete type. |

Author: I am thinking i can probably put limitation into a separate section like this 
<img width="789" alt="image" src="https://github.com/user-attachments/assets/7e868ba7-5035-4177-84de-a93e74f2885d">

CALL catalog_name.system.rewrite_data_files(table => 'db.sample', strategy => 'sort', sort_order => 'zorder(c1,c2)');
```

Rewrite the data files in table `db.sample` using bin-pack strategy in any partition where at least two files need rewriting, and then remove any dangling delete files.
```sql

> CALL catalog_name.system.rewrite_data_files(table => 'db.sample', options => map('min-input-files','2','remove-dangling-deletes', 'true'));

Reviewer: Do you think we can make spaces in the whole map for consistency?

Author: added space after all comma within map
